packages/nimble-components/src/table/components/row/template.ts

```diff
@@ -1,14 +1,33 @@
 import { html, ref, repeat, when } from '@microsoft/fast-element';
 import type { TableRow } from '.';
-import type { MenuButtonToggleEventDetail } from '../../../menu-button/types';
+import {
+    ButtonAppearance,
+    type MenuButtonToggleEventDetail
+} from '../../../menu-button/types';
 import { tableCellTag } from '../cell';
 import { checkboxTag } from '../../../checkbox';
-import { tableRowSelectLabel } from '../../../label-provider/table/label-tokens';
+import {
+    tableRowCollapseLabel,
+    tableRowExpandLabel,
+    tableRowSelectLabel
+} from '../../../label-provider/table/label-tokens';
 import type { TableColumn } from '../../../table-column/base';
+import { buttonTag } from '../../../button';
+import { iconArrowExpanderRightTag } from '../../../icons/arrow-expander-right';
+import {
+    controlHeight,
+    mediumPadding
+} from '../../../theme-provider/design-tokens';
 
 // prettier-ignore
 export const template = html<TableRow>`
-    <template role="row" aria-selected=${x => x.ariaSelected}>
+    <template 
+        role="row"
+        aria-selected=${x => x.ariaSelected}
+        style="--ni-private-table-row-spacer-width: ${x => (x.nestingLevel > 0 || (x.nestingLevel === 0 && !(x.rowHierarchyLevel === 'parent' || x.rowHierarchyLevel === 'topLevelParent'))
```

Comment:
This is very confusing to me. Ideally (i.e. if we can stop conditionally puting the expand/collapse button in the cell), I would expect this to be something like `x => x.nestingLevel * controlHeight.getValueFor(x)`. But at the very least, I think this would be a slight improvement: `x => (x.nestingLevel > 0 || x.rowHierarchyLevel === undefined || x.rowHierarchyLevel === 'leaf') ? ...`.

Reply:
Didn't need this after refactor.